### PR TITLE
Missing support for PartitionAware as in Java client

### PR DESCRIPTION
**Pull request by ihsandemir**
Implemented PartitionAware support.

### CODE REVIEW

**Comment by ihsandemir** [comment]
verify

File: hazelcast/include/hazelcast/client/ClientConfig.h [hunk 1]
```diff
@@ -296,7 +296,7 @@ namespace hazelcast {
             *
             *  @return serializationConfig
             */
-            SerializationConfig const &getSerializationConfig() const;
+            SerializationConfig &getSerializationConfig();
```
Comment (sancar) [inline hazelcast/include/hazelcast/client/ClientConfig.h]: What was the reason behind removing this two `const`'s ?

Comment (ihsandemir) [inline hazelcast/include/hazelcast/client/ClientConfig.h]: if it is const, then we can not do getSerializationConfig().registerSerializer(). The only way then to registerSerializer is the ClientConfig.setSerializationConfig(). This was the motivation behind it. I was to use it in a test but I later did not use it but I thought we better fix this as well.

**Comment by ihsandemir** [comment]
verify

**Pull request merged**